src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -1135,6 +1135,20 @@ void fightAirRaidsAndStrategicBombing(final IDelegateBridge delegateBridge, Supp
     }
   }
 
+  public void fightAutoKills(final IDelegateBridge aBridge) {
+    // Kill undefended transports. Done first to remove potentially dependent sea battles
+    // Which could block amphibious assaults below
+    getPendingBattleSites(false).stream().map(territory -> getPendingBattle(territory, false, BattleType.NORMAL))
+          .filter( battle -> Match.allMatch(battle.getDefendingUnits(), Matches.UnitIsDefenselessTransport))
+          .forEach( battle -> battle.fight(aBridge));
+    getPendingBattleSites(false).stream().map(territory -> getPendingBattle(territory, false, BattleType.NORMAL))
+          .forEach( battle -> {
+                 if (battle instanceof NonFightingBattle && getDependentOn(battle).isEmpty()) {
```

Comment:
Not in any way I'm aware of. IIRC the instanceof operator doesn't work in that context.